internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -155,6 +156,9 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 		if conditions.IsTrue(t.Cluster, clusterv1.InfrastructureReadyCondition) && clusterInfraReady != nil && clusterInfraReady.Time.After(comparisonTime) {
 			comparisonTime = clusterInfraReady.Time
 		}
```

Comment:
Q: So "nodeStartup" then doesn't include the time to get the InfraMachine to ready it would then be only the time for bootstrap on that machine?
(but that only if the InfraMachine becomes ready, if it never becomes ready nodeStartup includes InfraMachine creation?)

Reply:
Godoc of the field
> // Machines older than this duration without a node will be considered to have
> // failed and will be remediated.

This seems to hint towards machineCreationTime, but that's already not true today (as we check for controlPlaneInitializedCondition and InfrastructureReadyCondition)

Reply:
> (but that only if the InfraMachine becomes ready, if it never becomes ready nodeStartup includes InfraMachine creation?)

That's my understanding as well